### PR TITLE
[iDEAL] Redirect API Call

### PR DESCRIPTION
**Pull request by mpetrenco**

### CODE REVIEW

File: Source/API/JPSession.h [hunk 1]
```diff
@@ -80,4 +80,17 @@ typedef void (^JudoCompletionBlock)(JPResponse *_Nullable, NSError *_Nullable);
...
 
+/**
+ *  A general method for making REST requests
+ *
+ *  @param HTTPMethod the HTTP request method
+ *  @param path       the path
+ *  @param parameters information that is set in the HTTP Body
+ *  @param completion completion callblack block with the results
+ */
+- (void)requestWithMethod:(nonnull NSString *)HTTPMethod
```
Comment (stpopa) [inline Source/API/JPSession.h]: what's the point of having this if there's the 
`- (void)POST:(NSString *)path parameters:(NSDictionary *)parameters completion:(JudoCompletionBlock)completion` ?

File: Source/Managers/IDEALManager.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
+@class JPAmount, JPReference, IDEALBank, JPSession;
+
+@interface IDEALManager : NSObject
```
Comment (stpopa) [inline Source/Managers/IDEALManager.h]: let's rename this to `IDEALSerivice`

File: Source/Model/JPTransactionData.h [hunk 3]
```diff
@@ -94,6 +94,11 @@ typedef NS_ENUM(NSUInteger, TransactionResult) {
  */
 @property (nonatomic, assign) TransactionType type;
 
+/**
+ *  A redirect URL used for iDEAL bank transactions
+ */
+@property (nonatomic, assign) NSString *_Nullable redirectUrl;
```
Comment (stpopa) [inline Source/Model/JPTransactionData.h]: is it possible to avoid having the `redirectUrl` here?
let's discuss this offline please

Comment (mpetrenco) [reply Source/Model/JPTransactionData.h]: We will consider changing the logic in a future PR

**Pull request merged**